care/facility/tests/test_facility_api.py

```diff
@@ -206,86 +228,140 @@ def test_facility_list_retrieval(self):
+            - test status code 201 is returned
+        """
+        user = self.user
+        user.is_superuser = True
+        user.save()
+
+        # although this should be put, PUT is weirdly enough not allowed here
+        response = self.client.post(self.get_url(), self.get_list_representation(self.facility), format="json",)
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED)
 
-class TestFacilityBulkUpdate(APITestCase):
+
+class TestFacilityBulkUpdate(TestBase):
     """Test API for bulk updates to facility"""
 
-    @classmethod
-    def setUpTestData(cls):
-        # super(TestFacility, cls).setUpTestData()
-        cls.su_data = {
-            "user_type": 5,
-            "email": "some.email@somedomain.com",
-            "phone_number": "5554446667",
-            "age": 30,
-            "gender": 2,
-            "district": 11,
-            "username": "superuser_1",
-            "password": "bar",
-        }
-        cls.super_user = User.objects.create_superuser(**cls.su_data)
-        cls.user_data = cls.su_data.copy()
-        cls.user_data["username"] = "user"
-        cls.user = User.objects.create_user(**cls.user_data)
-        cls.facility_data = {
-            "name": "Foo",
-            "district": 13,
-            "facility_type": 1,
-            "address": "8/88, 1st Cross, 1st Main, Boo Layout",
-            "location": {"latitude": 49.878248, "longitude": 24.452545},
-            "oxygen_capacity": 10,
-            "phone_number": "9998887776",
+    def get_base_url(self):
+        return "/api/v1/facility/bulk_upsert"
+
+    def get_list_representation(self, facility: Any = None) -> dict:
+
+        if isinstance(facility, dict):
+            location = facility.pop("location", {})
+            district_id = facility.pop("district")
+            facility = Facility(
+                **facility, district_id=district_id, location=Point(location["longitude"], location["latitude"]),
+            )
+
+        return {
+            "name": facility.name,
+            "district": facility.district.id,
+            "facility_type": 3,
+            "address": facility.address,
+            "capacity": [{"room_type": 1, "total_capacity": 100, "current_capacity": 48}],
         }
-        # **cls.data is not used because of a issue with the location attribute which requires a Point object
-        cls.facility = Facility.objects.create(
-            name="Foo",
-            district=13,
-            facility_type=1,
-            address="8/88, 1st Cross, 1st Main, Boo Layout",
-            location=Point(24.452545, 49.878248),
-            oxygen_capacity=10,
-            phone_number="9998887776",
-        )
 
-    def setUp(self):
-        """This is run before every class method"""
-        self.client.login(username=self.su_data["username"], password=self.su_data["password"])
+    def get_detail_representation(self, facility: Facility) -> dict:
+        return {
```

Comment:
I don't think these were required in this class because it is not used. You could have just did a `pass`.